base/boot.jl

```diff
@@ -221,10 +221,10 @@ end
 
 abstract DirectIndexString <: AbstractString
 
-immutable String <: AbstractString
-    data::Array{UInt8,1}
-    # required to make String("foo") work (#15120):
-    String(d::Array{UInt8,1}) = new(d)
+type String <: AbstractString
```

Comment:
I assume you addressed the aforementioned "problems" including the special `deepcopy` implementation?